rusty_pizza_server/src/order_model/meal.rs

```diff
@@ -44,24 +62,37 @@ impl Meal {
             meal_id,
             variety,
             price,
-            specials: BTreeSet::new(),
+            specials: HashMap::new(),
+            special_factory: SpecialFactory::new(),
         }
     }
 
     pub fn get_id(&self) -> u32 {
         self.id
     }
 
-    pub fn add_special(&mut self, special: String) {
-        self.specials.insert(special);
+    pub fn get_price(&self) -> Money {
+        self.price
     }
 
-    pub fn remove_special(&mut self, special: &String) {
-        self.specials.remove(special);
+    /// Creates and adds a new special and returns a mutable reference to it.
```

Comment:
A more structered rustdoc would be nice:

/// Header
///
/// # Arguments
///
/// [List of arguments]
///
/// # Return
///
/// Return type and description

Reply:
I don't actually like this documentation style, having to document all arguments and the return type often duplicates information. Maybe the return type description is a bit short but I generally like the style of the official documentation (e.g. [for the `insert` method of `HashMap`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.insert)).